### PR TITLE
Per institution LoA configuration not working

### PR DESCRIPTION
**Pull request by MKodde**
This pull request should fix the problem that an institution specific LoA is not handled correctly by the Gateway. The pivotal story goes into more detail about the expected behaviour.

Tasks that have been performed on this PR are:

- [x] Fix the bug
- [x] Investigate why the `authenticatingIdp` is not passed to the `resolveHighestRequiredLoa` method call.

[Pivotaltracker #149563137](https://www.pivotaltracker.com/story/show/149563137)

### CODE REVIEW

File: src/Surfnet/StepupGateway/GatewayBundle/Service/InstitutionMatchingHelper.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
```
Comment (pmeulen) [inline src/Surfnet/StepupGateway/GatewayBundle/Service/InstitutionMatchingHelper.php]: 2014?

Comment (MKodde) [inline src/Surfnet/StepupGateway/GatewayBundle/Service/InstitutionMatchingHelper.php]: Fixedup!

File: src/Surfnet/StepupGateway/GatewayBundle/Service/StepUpAuthenticationService.php [hunk 2]
```diff
@@ -157,26 +165,21 @@ public function resolveHighestRequiredLoa(
...
-            );
+        $matchingInstitutions = $this->institutionMatchingHelper->findMatches(
+            array_keys($spConfiguredLoas),
+            $institutions
+        );
 
-            if (array_key_exists($serviceProvider->getEntityId(), $idpConfiguredLoas)) {
-                $loaCandidates->add($idpConfiguredLoas[$serviceProvider->getEntityId()]);
+        if (count($matchingInstitutions) > 0) {
+            $this->logger->info(sprintf('Found matching SP configured LoA\'s'));
```
Comment (jorissteyn) [inline src/Surfnet/StepupGateway/GatewayBundle/Service/StepUpAuthenticationService.php]: Why sprintf?

File: src/Surfnet/StepupGateway/GatewayBundle/Service/InstitutionMatchingHelper.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+     *
+     * Returns the match found in the $institutionList1.
+     *
+     * @param array $institutionList1
+     * @param array $institutionList2
+     * @return array
+     */
+    public function findMatches(array $institutionList1, array $institutionList2)
+    {
+        $matchingInstitutions = array_values(array_uintersect($institutionList1, $institutionList2, 'strcasecmp'));
```
Comment (jorissteyn) [inline src/Surfnet/StepupGateway/GatewayBundle/Service/InstitutionMatchingHelper.php]: Feels a bit overkill to have an object just for calculating the intersection.

**Comment by pmeulen** [comment]
I just finished testing a few scenarios in test, and everything to be working as it should now. Thanks for the quick turnaround on the fix!

**Pull request merged**